src/Core/Injector/MethodInjector.php

```diff
@@ -0,0 +1,108 @@
+<?php
+namespace SilverStripe\Core\Injector;
+
+use ReflectionMethod;
+
+/**
+ * Parses method reflections and attempts to auro-wire parameters for injection
+ */
+class MethodInjector
+{
+    const TAG_INJECTABLE = 'Injectable';
+    const TAG_ATTRIBUTE = 'Inject';
+
+    /**
+     * @var ReflectionMethod
+     */
+    protected $method;
+
+    /**
+     * MethodInjector constructor.
+     * @param ReflectionMethod $method
+     */
+    public function __construct(ReflectionMethod $method)
+    {
+        $this->method = $method;
+    }
+
+    /**
+     * Indicates that the method is marked as injectable in the doc-comment for the method (with @Injectable)
+     *
+     * @return bool
+     */
+    public function isMarkedInjectable()
```

Comment:
I'll update to `provideParams` - nice catch. I prefer `isMarkedInjectable` because a user could still attempt to inject a method that isn't marked as "injectable".

Reply:
Yeah good point